sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Customizations/AcrClientCredentials.cs

```diff
@@ -0,0 +1,347 @@
+﻿using Microsoft.IdentityModel.Tokens;
+using Microsoft.Rest;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.ContainerRegistry
+{
+    /*Robust handling of Basic and OAUTH2 authentication flows for the Azure Container Registry Runtime .Net SDK.
```

Comment:
>/* [](start = 4, length = 2)

NIT: use /// for comments. These are all minor C# specific guidelines so running the code formatter might help fix most of these.